### PR TITLE
Use DeferredRegister in MSBlocks

### PR DESCRIPTION
**Pull request by Dweblenod**

### CODE REVIEW

File: src/main/generated/resources/data/minestuck/minestuck/grist_cost_generation_recipes.json [hunk 1]
```diff
@@ -19,11 +19,6 @@
     "source": "stonecutting",
     "interpreter_type": "minestuck:default"
   },
-  {
-    "source_type": "recipe_type",
-    "source": "smithing",
-    "interpreter_type": "minestuck:smithing"
-  },
```
Comment (kirderf1) [inline src/main/generated/resources/data/minestuck/minestuck/grist_cost_generation_recipes.json:26]: Can you revert this change? (Or alternatively fix it in the data provider, though that's already being done in #359)

File: src/main/java/com/mraof/minestuck/block/MSBlocks.java [hunk 2]
```diff
@@ -19,681 +19,518 @@
...
+	public static final RegistryObject<Block> SKAIA_PORTAL = REGISTER.register("skaia_portal", () -> new SkaiaPortalBlock(Block.Properties.of(Material.PORTAL, MaterialColor.COLOR_CYAN).noCollission().lightLevel(state -> 11).strength(-1.0F, 3600000.0F).noDrops()));
+	static BlockBehaviour.Properties blackChessBricks = Block.Properties.of(Material.STONE, MaterialColor.TERRACOTTA_BLACK).requiresCorrectToolForDrops().strength(1.5F, 6.0F);
+	static BlockBehaviour.Properties darkGrayChessBricks = Block.Properties.of(Material.STONE, MaterialColor.TERRACOTTA_GRAY).requiresCorrectToolForDrops().strength(1.5F, 6.0F);
+	static BlockBehaviour.Properties lightGrayChessBricks = Block.Properties.of(Material.STONE, MaterialColor.TERRACOTTA_LIGHT_GRAY).requiresCorrectToolForDrops().strength(1.5F, 6.0F);
+	static BlockBehaviour.Properties whiteChessBricks = Block.Properties.of(Material.STONE, MaterialColor.TERRACOTTA_WHITE).requiresCorrectToolForDrops().strength(1.5F, 6.0F);
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/block/MSBlocks.java:49]: I'd suggest cleaning these up, renaming them, making them private static final, and potentially writing some of these as helper functions instead. This applies to pretty much anything here that's not a registry object or the deferred register.
I'd also like you to be adding empty newlines to split things up more.

File: src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java [hunk 3]
```diff
@@ -64,435 +64,435 @@ public class MinestuckEnUsLanguageProvider extends MinestuckLanguageProvider
...
-		add(MSBlocks.PUSHABLE_BLOCK, "Pushable Block");
-		add(MSBlocks.AND_GATE_BLOCK, "AND Gate");
-		add(MSBlocks.OR_GATE_BLOCK, "OR Gate");
-		add(MSBlocks.XOR_GATE_BLOCK, "XOR Gate");
-		add(MSBlocks.NAND_GATE_BLOCK, "NAND Gate");
-		add(MSBlocks.NOR_GATE_BLOCK, "NOR Gate");
-		add(MSBlocks.XNOR_GATE_BLOCK, "XNOR Gate");
-		add(MSBlocks.GATE, "Gate");
-		add(MSBlocks.RETURN_NODE, "Return Node");
+		add(MSBlocks.BLACK_CHESS_DIRT.get(), "Black Chess Tile");
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java:67]: Use `addBlock()` instead.
Example:
`addBlock(MSBlocks.BLACK_CHESS_DIRT, "Black Chass Tile");`

File: src/main/java/com/mraof/minestuck/item/MSItems.java [hunk 4]
```diff
@@ -502,403 +502,403 @@ public static void registerItems(final RegistryEvent.Register<Item> event)
...
-		registerItemBlock(registry, BLACK_CHESS_DIRT, MSItemGroup.MAIN);
-		registerItemBlock(registry, WHITE_CHESS_DIRT, MSItemGroup.MAIN);
-		registerItemBlock(registry, DARK_GRAY_CHESS_DIRT, MSItemGroup.MAIN);
-		registerItemBlock(registry, LIGHT_GRAY_CHESS_DIRT, MSItemGroup.MAIN);
-		registerItemBlock(registry, SKAIA_PORTAL, new Item.Properties().tab(MSItemGroup.MAIN).rarity(Rarity.EPIC));
+		registerItemBlock(registry, BLACK_CHESS_DIRT.get(), MSItemGroup.MAIN);
+		registerItemBlock(registry, WHITE_CHESS_DIRT.get(), MSItemGroup.MAIN);
+		registerItemBlock(registry, DARK_GRAY_CHESS_DIRT.get(), MSItemGroup.MAIN);
+		registerItemBlock(registry, LIGHT_GRAY_CHESS_DIRT.get(), MSItemGroup.MAIN);
+		registerItemBlock(registry, SKAIA_PORTAL.get(), new Item.Properties().tab(MSItemGroup.MAIN).rarity(Rarity.EPIC));
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/item/MSItems.java:509]: I'd suggest that you instead update `registerItemBlock()` to take a `Supplier<Block>`

File: src/main/java/com/mraof/minestuck/world/gen/feature/tree/EndTreeFeature.java [hunk 5]
```diff
@@ -32,7 +33,7 @@ public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context)
 		WorldGenLevel level = context.level();
 		BlockPos position = context.origin();
 		BlockPos soilPos = position.below();
-		if(level.getBlockState(soilPos).canSustainPlant(level, soilPos, Direction.UP, MSBlocks.END_SAPLING))
+		if(level.getBlockState(soilPos).canSustainPlant(level, soilPos, Direction.UP, (IPlantable) MSBlocks.END_SAPLING.get()))
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/feature/tree/EndTreeFeature.java:36]: Remove the need for this new cast by changing the type of `END_SAPLING` to `RegistryObject<BushBlock>` to match the block type that `END_SAPLING` had before.

**Review by kirderf1 — CHANGES_REQUESTED** [review]
Any thoughts on the multiblock classes and if they should be updated in any way accordingly?

File: src/main/java/com/mraof/minestuck/item/MSItems.java (hunk 4 above)
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/item/MSItems.java:509]: Not sure. I suppose it doesn't matter that much as long as that is on the way.

File: src/main/java/com/mraof/minestuck/block/MSBlocks.java (hunk 2 above)
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/block/MSBlocks.java:49]: Helper function for streamlining which portion? Working on the rest

File: src/main/java/com/mraof/minestuck/world/gen/feature/tree/EndTreeFeature.java (hunk 5 above)
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/world/gen/feature/tree/EndTreeFeature.java]: fixed

File: src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java (hunk 3 above)
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/data/MinestuckEnUsLanguageProvider.java]: Now have done so. Also changed addTooltip to addBlockTooltip, addExtra to addBlockExtra, etc

File: src/main/java/com/mraof/minestuck/block/MSBlocks.java (hunk 2 above)
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/block/MSBlocks.java]: I was thinking of a helper function that creates and returns some block property, as opposed to creating a block property statically. I don't think I had any specific in mind.
Though now that I've given it some more thought, I realize that that because block properties are mutable, there's a point in making all these common properties provided by helper functions.

**Review by kirderf1 — APPROVED** [review]
